internal/backend/storage.go

```diff
@@ -49,22 +49,23 @@ type Storage interface {
 	Fsck(context.Context) error
 }
 
-// DetectStorage tries to detect the storage backend being used
+// DetectStorage tries to detect the storage backend being used.
 func DetectStorage(ctx context.Context, path string) (Storage, error) {
-	if HasStorageBackend(ctx) {
-		if be, err := StorageRegistry.Get(GetStorageBackend(ctx)); err == nil {
-			st, err := be.New(ctx, path)
-			if err == nil {
-				return st, nil
-			}
-			be, err := StorageRegistry.Get(FS)
-			if err != nil {
-				return nil, err
-			}
-			return be.Init(ctx, path)
+	// The call to HasStorageBackend is important since GetStorageBackend will always return FS
+	// if nothing is found in the context.
+	if be, err := StorageRegistry.Get(GetStorageBackend(ctx)); HasStorageBackend(ctx) && err == nil {
```

Comment:
Could it be that the weird `clone` behaviour where it's not properly initializing the Git config because it's probably detecting the file system as being `fs` instead of `gitfs` be related to that part and the below call to Init? See the issue in my comment: https://github.com/gopasspw/gopass/issues/1162#issuecomment-998829861